containers/ecr-viewer/src/app/view-data/utils/fhir-data-utils.ts

```diff
@@ -0,0 +1,72 @@
+import { FhirResource, Period } from "fhir/r4";
+
+import { evaluateOne } from "@/app/utils/evaluate";
+import { FhirPath, PathTypes } from "@/app/utils/evaluate/fhir-paths";
+
+const ONGOING_DATE = new Date("9999-01-01");
+
+const parseDate = (dateString: string | undefined): Date | undefined =>
+  dateString ? new Date(dateString) : undefined;
+
+const getDates = <K extends keyof PathTypes>(
+  obs: FhirResource,
```

Comment:
given this could be a non-observation, can we change the variable name to something like `resource`?